env/src/main/java/com/sap/cloud/security/config/k8s/K8sEnvironment.java

```diff
@@ -48,15 +55,58 @@ public Type getType() {
 		return Type.KUBERNETES;
 	}
 
+	private void loadAll() {
+		Map<String, OAuth2ServiceConfiguration> xsuaaPlans = DefaultServiceBindingAccessor.getInstance().getServiceBindings().stream()
+				.filter(b -> XSUAA.getCFName().equalsIgnoreCase(b.getServiceName().orElse(null)))
+				.map(b -> mapToOAuth2ServiceConfiguration(b)).collect(Collectors.toMap(c -> c.getProperty("plan"),
+						Function.identity()));
+		Map<String, OAuth2ServiceConfiguration> identityPlans = DefaultServiceBindingAccessor.getInstance().getServiceBindings().stream()
+				.filter(b -> IAS.getCFName().equalsIgnoreCase(b.getServiceName().orElse(null)))
+				.map(b -> mapToOAuth2ServiceConfiguration(b)).collect(Collectors.toMap(c -> c.getProperty("plan"),
+						Function.identity()));
+		serviceConfigurations.put(XSUAA, xsuaaPlans);
+		serviceConfigurations.put(IAS, identityPlans);
+	}
+
+	private OAuth2ServiceConfiguration mapToOAuth2ServiceConfiguration(ServiceBinding b) {
+		if (!b.getServiceName().isPresent()) {
+			LOGGER.error("Ignores Service Binding with name {} as service name is not provided.", b.getName());
+			return null; // TODO test
+		}
+		final Service service = from(b.getServiceName().get());
+		OAuth2ServiceConfigurationBuilder configBuilder =  OAuth2ServiceConfigurationBuilder.forService(service)
+				.withProperties(TypedMapView.ofCredentials(b).getEntries(String.class))
```

Comment:
> `TypedMapView.ofCredentials(b).getEntries(String.class)`

I fear this might not be doing what you would expect it to do...
This method will **filter** the underlying `Map<String. Object>` (the "raw" credentials) for entries that match the queried type (`String` in this case).
All other entries (such as numbers, booleans, Lists and complex Objects) will **not** be returned.

----

I'm generally not so happy about how the `TypedMapView` needs to be used here.
I think we should probably add some compatibility methods such as `.toMap` and `.toList` to the API.
Otherwise things are just too cumbersome.

What do you think, @nenaraab?